### DISCUSSION TITLE
Windows 11 GPU and NPU activity

### DISCUSSION THREAD

**Discussion by RockyMtnBri** [community/community#137087] [Programming Help]
I'm trying to write a simple Jupyter Notebook that includes a graphical representation of the CPU, GPU, and NPU on the laptop it's running on. The CPU information was easy to grab via psutil, but I can't find something similar for the GPU (Intel integrated) and NPU. Can anyone assist? Here's the code I created via Copilot (note the GPU and NPU placeholders):

# Import necessary libraries
import psutil
import matplotlib.pyplot as plt
import time
from IPython.display import clear_output

# Function to get CPU, GPU, and NPU usage
def get_system_usage():
    cpu_usage = psutil.cpu_percent(interval=1)
    memory_info = psutil.virtual_memory()
    memory_usage = memory_info.percent
    # Placeholder for GPU and NPU usage
    gpu_usage = 0  # Replace with actual GPU usage retrieval
    npu_usage = 0  # Replace with actual NPU usage retrieval
    return cpu_usage, memory_usage, gpu_usage, npu_usage

# Function to plot system usage
def plot_usage(cpu_usage, memory_usage, gpu_usage, npu_usage):
    labels = ['CPU', 'Memory', 'GPU', 'NPU']
    usage = [cpu_usage, memory_usage, gpu_usage, npu_usage]
    
    plt.figure(figsize=(10, 5))
    plt.bar(labels, usage, color=['blue', 'green', 'red', 'purple'])
    plt.ylim(0, 100)
    plt.xlabel('Component')
    plt.ylabel('Usage (%)')
    plt.title('System Usage')
    plt.show()

# Main loop to update and display system usage
while True:
    cpu_usage, memory_usage, gpu_usage, npu_usage = get_system_usage()
    clear_output(wait=True)
    plot_usage(cpu_usage, memory_usage, gpu_usage, npu_usage)
    time.sleep(1)

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @RockyMtnBri ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by RockyMtnBri** [reply]
Thanks!

**Comment by RockyMtnBri** [comment]
Not seeing any responses.
